### PR TITLE
4750 Moves selectMode to redux store

### PR DESCRIPTION
**Pull request by rmhowe**
# Description

Continuing on from [this original PR](https://github.com/medic/medic/pull/5323) (followed by [this PR](https://github.com/medic/medic/pull/5368)) moving scope properties across to the redux store.

medic/medic#4750

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

File: webapp/src/js/controllers/reports-content.js [hunk 1]
```diff
@@ -21,7 +21,12 @@ var _ = require('underscore');
       'ngInject';
 
       var ctrl = this;
-      var unsubscribe = $ngRedux.connect(null, Actions)(ctrl);
+      var mapStateToTarget = function(state) {
+        return {
+          selectMode: state.selectMode
+        };
+      };
+      var unsubscribe = $ngRedux.connect(mapStateToTarget, Actions)(ctrl);
```
Comment (dianabarsan) [inline webapp/src/js/controllers/reports-content.js]: I think we should already start mapping dispatch to target, even though we only have 6 actions so far.
(applies to all controllers that are currently allowed to call all actions).

Comment (rmhowe) [reply webapp/src/js/controllers/reports-content.js]: Sounds good, I think it's a bit convoluted when using `ng-redux` but it should be fine with thunks

**Comment by rmhowe** [comment]
@dianabarsan so I've been using the redux devtools extension but haven't committed any of the code to make it work in the app, in my opinion the devtools are better than logger middleware but I'm happy to put the logger in as well! Especially if people don't want to install the extension

File: webapp/src/js/controllers/reports-content.js (hunk 1 above)
Comment (dianabarsan) [reply webapp/src/js/controllers/reports-content.js]: Convoluted? I got it working with this (for the super simplistic reducers that we currently have):
```suggestion
var mapDispatchToTarget = function(dispatch) {
  return {
    clearCancelCallback: Actions(dispatch).clearCancelCallback
  }
};
```
Re-reading this, I guess it's a bit convoluted and unnecessary at this point. 
About thunks, I imagine we are going to need them at some point anyway.

Comment (rmhowe) [reply webapp/src/js/controllers/reports-content.js]: Yep that's the approach I was thinking of too, no I think you're right it's good to add them in now it makes the logic flow a bit clearer. We probably will need thunks eventually but I guess it's best to leave adding the library until we need it so I'll just stick with this for now

**Pull request merged**